File: frontend/src/app/domain/responses/profile-response.ts
import {ImageObject} from '../objects/image-object';
import {ExternalURLObject} from '../objects/external-urlobject';
import {FollowersObject} from '../objects/followers-object';

export interface ProfileResponse {
  country: string;
  display_name: string;
  email: string;
  explicit_content: object;
  external_urls: ExternalURLObject;
  followers: FollowersObject;
  href: string;
  id: string;
  images: ImageObject[];
  product: string;
  type: string;
  uri: string;
}

File: backend/src/main/java/organizer/backend/domain/responses/ProfileResponse.java
package organizer.backend.domain.responses;

import organizer.backend.domain.objects.ExplicitContentObject;
import organizer.backend.domain.objects.ExternalURLObject;
import organizer.backend.domain.objects.FollowersObject;
import organizer.backend.domain.objects.ImageObject;

public record ProfileResponse(
        String country,
        String display_name,
        String email,
        ExplicitContentObject explicit_content,
        ExternalURLObject external_urls,
        FollowersObject followers,
        String href,
        String id,
        ImageObject[] images,
        String product,
        String type,
        String uri) {
}
